### ISSUE TITLE
Can't propogate trace context with http instrumentation?

### ISSUE THREAD

**Issue by yantang-msft** [census-instrumentation/opencensus-node#117]
I'm playing with this library and see how the correlation context is propagated through http request.
Here is the snippet
``` node
const tracing = require('@opencensus/nodejs');
const ConsoleExporter = require('@opencensus/core');
const propagation = require('@opencensus/propagation-tracecontext');

const traceContext = new propagation.TraceContextFormat();

const tracer = tracing.start({propagation: traceContext}).tracer;
const exporter = new ConsoleExporter.ConsoleExporter({
    bufferSize: 1,
    bufferTimeout: 1,
    logger: console
});
tracing.registerExporter(exporter);

const http = require('http');

http.get('http://www.google.com', (resp) => {
    let data = '';
    resp.on('data', (chunk) => {
        data += chunk;
    });

    resp.on('end', () => {
        console.log(`Response end, data length is ${data.length}`);
    });
}).on("error", (err) => {
    console.log("Error: " + err.message);
});

const stdin = process.openStdin();
stdin.addListener("data", () => {});
```

If I don't add the propagation context, the http instrumentation would successfully create a span for the outgoing request. But if I add the propagation context, it gives me this error. Did I missed anything?
```
Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at ClientRequest.setHeader (_http_outgoing.js:469:11)
    at Object.setHeader (D:\OpenCensus\node\node_modules\@opencensus\instrumentation-http\build\src\http.js:182:29)
    at TraceContextFormat.inject (D:\OpenCensus\node\node_modules\@opencensus\propagation-tracecontext\build\src\tracecontext-format.js:109:20)
    at D:\OpenCensus\node\node_modules\@opencensus\instrumentation-http\build\src\http.js:187:29
    at contextManager.runAndReturn (D:\OpenCensus\node\node_modules\@opencensus\core\build\src\trace\model\tracer.js:111:20)
    at AsyncHooksNamespace.runAndReturn (D:\OpenCensus\node\node_modules\@opencensus\core\build\src\internal\cls-ah.js:51:21)
    at CoreTracer.startRootSpan (D:\OpenCensus\node\node_modules\@opencensus\core\build\src\trace\model\tracer.js:84:36)
    at Object.outgoingRequest [as get] (D:\OpenCensus\node\node_modules\@opencensus\instrumentation-http\build\src\http.js:159:42)
    at Object.<anonymous> (D:\OpenCensus\node\test.js:17:6)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
```

**Comment by vivasaayi** [comment]
I can reproduce the same error. This happens when a new HTTP request is made by the service. 

There is no RootSpan in this case and a new Root Span is created. Some more logs from me:

```
[debug] "2018-09-11T14:04:31.722Z"  'http plugin outgoingRequest'
[debug] "2018-09-11T14:04:31.722Z"  'outgoingRequest starting a root span'
[debug] "2018-09-11T14:04:31.723Z"  'starting RootSpan  { traceId: \'8ca78b878e944ec197eba66c83ec9d79\',\n  id: \'268494413949085\',\n  parentSpanId: null,\n  traceState: undefined }'
[debug] "2018-09-11T14:04:31.723Z"  'starting to notify listeners the start of rootspans'
[debug] "2018-09-11T14:04:31.723Z"  'makeRequestTrace'
INJECT ERROR: Error: Can't set headers after they are sent.
      at validateHeader (_http_outgoing.js:491:11)
      at ClientRequest.setHeader (_http_outgoing.js:498:3)
      at Object.setHeader (/srv/package/node_modules/@opencensus/instrumentation-http/build/src/http.js:183:29)
```

**Comment by mayurkale22** [comment]
Closed in #324, Will have an example for this soon #369 (PR in review)
